Guide for Web Key Directory

I'd like to share my notes how i set up a Web Key Directory on U6 (but I think there shouldn't be any differences for U7). I currently do not have time to translate this to ReST, but maybe someone else has.


https://spacekookie.de/blog/usable-gpg-with-wkd/
> [WKD] stands for "Web Key Directory" and is a standard for making a users public key available via their e-mail provider or server with the domain that corresponds to their e-mail address. There's several clients (such as Enigmail in Thunderbird) that will use this standard to automatically fetch a user's public key, when writing an e-mail to them.

See also https://wiki.gnupg.org/WKD .

# Create Directory Structure
```bash
mkdir -p ~/html/.well-known/openpgpkey/hu
touch ~/html/.well-known/openpgpkey/policy
```
~~Allow Directory Access:~~
DO NOT DO THIS:
~~echo "Options +Indexes" >> \~/html/.well-known/openpgpkey/hu/.htaccess~~
(can be a privacy issue, see Starkstromkonsuments comment below)

Instead use
```plaintext
## WEB KEY DIRECTORY ##
<IfModule mod_mime.c>
   ForceType application/octet-stream
   Header always set Access-Control-Allow-Origin "*"
</IfModule>
```
in `~/html/.well-known/openpgpkey/hu/.htaccess`.

# Calculate WKD-Hash of Mailuser
Take the prefix of your e-mail address (i.e. in `isabell@stardust.uberspace.de`, this would be `isabell`), hash it with SHA-1 and then encode the output with z-base-32.

You can use https://cryptii.com/pipes/z-base-32 for that (add a hash-block with SHA-1 before the z.base-32-block).
* E-Mail: isabell@startdust.uberspace.de
* Mailuser: isabell
* WKD-Hash: mmuhurigesr7z8hzf6sh5cmfsnmiiyyr

Alternatively you can invoke gpg with the option --with-wkd-hash on the machine where your keys reside:
```bash
gpg --with-wkd-hash --list-public-keys "isabell@stardust.uberspace.de"
[...]
      0123456789ABCDEF0123456789ABCDEF01234567
uid        [uneingeschränkt] Isabell <sabell@stardust.uberspace.de>
           mmuhurigesr7z8hzf6sh5cmfsnmiiyyr@stardust.uberspace.de
```

# Exporting and storing public GPG key

Export the **binary** Version of your public key into a file named as the WKD-hash calculated before:
```bash
gpg --export 0123456789ABCDEF0123456789ABCDEF01234567 > mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
scp mmuhurigesr7z8hzf6sh5cmfsnmiiyyr isabell@stardust.uberspace.de:/home/isabell/html/.well-known/openpgpkey/hu
```

# Test
```bash
env GNUPGHOME=$(mktemp -d) gpg --locate-keys isabell@stardust.uberspace.de
```

I gave it a try today, but i can't get sphinx to work on my machine to test it :-(

```
$ make html
Sphinx v3.0.3 in Verwendung

Extension error:
Could not import extension authorship (exception: No module named 'authorship')
Makefile:23: recipe for target 'html' failed
make: *** [html] Error 2
```
Does anybody have a hint for me?

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -6,17 +6,15 @@
 .. tag:: mail
 .. tag:: self-hosting
 
-#######
-Web Key Directory (WKD) for GnuPG
-#######
+#######################
+Web Key Directory (WKD)
+#######################
 
 .. tag_list::
 
-[WKD] stands for "`Web Key Directory`_" and is a standard for making a users public key available via their e-mail provider or server with the domain that corresponds to their e-mail address. There's several clients (such as Enigmail in Thunderbird) that will use this standard to automatically fetch a user's public key, when writing an e-mail to them.
+[WKD] stands for "`Web Key Directory`_" and is a standard for making a users GnuPG / OpenPGP public key available via their e-mail provider or server with the domain that corresponds to their e-mail address. There's several clients (such as Enigmail in Thunderbird or OpenKeyChain on Android) that will use this standard to automatically fetch a user's public key, when writing an e-mail to them.
 
-Web Key Directories provide an easy way to discover public keys through HTTPS. They provide an important piece to the infrastructure to improve the user experience for exchanging secure emails and files.
-
-In contrast to the public keyservers a Web Key Directory does not publish mail addresses. And it is an authoritative pubkey source for its domain.
+Web Key Directories provide an easy way to discover public keys through HTTPS. They provide an important piece to the infrastructure to improve the user experience for exchanging secure emails and files. In contrast to the public keyservers a Web Key Directory does not publish mail addresses and it is an authoritative pubkey source for its domain.
 
 ----
 
@@ -28,7 +26,7 @@ In contrast to the public keyservers a Web Key Directory does not publish mail a
 Prerequisites
 =============
 
-You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@stardust.uberspace.de).
+You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@stardust.uberspace.de). This guide does not explain how to generate GnuPG / OpenPGP keys.
 
 The domain for the advanced method has to be set up at your DNS-Hoster and within your uberspace:
 
@@ -42,7 +40,7 @@ The domain for the advanced method has to be set up at your DNS-Hoster and withi
 Create Directory Structure
 ==========================
 
-advanced method
+Advanced method
 ---------------
 
 The "advanced" method seems to be queried in front of the "direct" method by clients. If the subdomain openpgpkey resolves to a valid IP but the host at this IP does not provide the WKD (e.g. because of a wildcard in the DNS-Records), the WKD-lookup may fail.
@@ -57,9 +55,9 @@ Create the directories and the empty files:
  [isabell@stardust isabell]$ touch openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/.htaccess
  [isabell@stardust ~]$
 
-Add the following lines to the .htaccess-file to set the correct headers:
+Add the following lines to the empty ``.htaccess`` file to set the correct headers:
 
-.. code-block:: apache2
+.. code-block:: apache
 
  ## WEB KEY DIRECTORY ##
  <IfModule mod_mime.c>
@@ -70,7 +68,7 @@ Add the following lines to the .htaccess-file to set the correct headers:
 .. warning:: Make sure that there is no automatic directory listing! It is not necessary and it reveals the number of emailadresses (and their hashes) in the WKD (this can be a privacy issue)!
 
 
-direct method
+Direct method
 -------------
 
 Create the directories and symlinks
@@ -95,24 +93,23 @@ Upload GPG keys
 Obtaining the WKD-Hashes
 ------------------------
 
-The keys are stored in files named by the WKD-Hashes of the Mailuser. Take the prefix of your e-mail address (i.e. in isabell@stardust.uberspace.de, this would be isabell), hash it with SHA-1 and then encode the output with z-base-32.
+The keys are stored in files named by the WKD-Hashes of the Mailuser. Take the prefix of your e-mail address (i.e. in isabell@stardust.uberspace.de, this would be `isabell`), hash it with SHA-1 and then encode the output with z-base-32.
 
 You can get the hashes for all identities of your GPG key with this command:
 
 ::
 
- someuser@somehost ~]$ gpg --with-wkd-hash --list-public-keys "isabell@stardust.uberspace.de"
+ [someuser@somehost ~]$ gpg --with-wkd-hash --list-public-keys "isabell@stardust.uberspace.de"
  [...]
        0123456789ABCDEF0123456789ABCDEF01234567
- uid        [uneingeschränkt] Isabell <isabell@stardust.uberspace.de>
+ uid           [ unknown] Isabell <isabell@stardust.uberspace.de>
             mmuhurigesr7z8hzf6sh5cmfsnmiiyyr@stardust.uberspace.de
  [...]
- someuser@somehost ~]$
+ [someuser@somehost ~]$
 
 Alternatively you can use https://cryptii.com/pipes/z-base-32 for that (add a hash-block with SHA-1 before the z.base-32-block).
 
 Summary:
-
  * E-Mail: isabell@startdust.uberspace.de
  * Prefix: isabell
  * WKD-Hash: mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
@@ -121,12 +118,12 @@ Summary:
 Exporting a GPG key
 -------------------
 
-.. note:: The public key is the the same for all identities of a key. You can simply duplicate the exported key, using the corresponding WKD-Hash as filename.
-
 ::
 
  [someuser@somehost ~]$ gpg --no-armor --export isabell@stardust.uberspace.de > mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
  [someuser@somehost ~]$
+ 
+.. note:: The public key is the the same for all identities of a key. You can simply duplicate the exported key, using the corresponding WKD-Hash as filename.
 
 Upload
 ------
@@ -139,7 +136,34 @@ Upload
 Testing
 =======
 
-You can test the download by running:
+First of all, these two URLs should be available using your browser:
+
+ * Advanced method: https://openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+ * Direct method: https://stardust.uberspace.de/.well-known/openpgpkey/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+
+or ``curl``:
+
+.. code-block:: console
+ :emphasize-lines: 4,6
+
+ [someuser@somehost ~]$ curl -I https://openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+ HTTP/2 200 
+ date: Sat, 02 May 2020 19:16:17 GMT
+ content-type: application/octet-stream
+ content-length: 5298
+ access-control-allow-origin: *
+ last-modified: Mon, 13 Apr 2020 18:15:20 GMT
+ etag: "14b2-5a33010e34bb7"
+ accept-ranges: bytes
+ server: nginx
+ referrer-policy: strict-origin-when-cross-origin
+ strict-transport-security: max-age=172800
+ x-content-type-options: nosniff
+ x-xss-protection: 1; mode=block
+ x-frame-options: SAMEORIGIN
+ [someuser@somehost ~]$ 
+
+You can test the WKD-download by running:
 
 ::
 
@@ -163,7 +187,6 @@ Credits
  * https://wiki.gnupg.org/WKDHosting
  * https://spacekookie.de/blog/usable-gpg-with-wkd/
  * https://www.kuketz-blog.de/gnupg-web-key-directory-wkd-einrichten/ (German)
-
 
 .. _Web Key Directory: https://wiki.gnupg.org/WKD
 

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -49,7 +49,7 @@ Create the directories and the empty files:
 
 ::
 
- [isabell@stardust ~]$ cd /var/www/virtual/isabell/
+ [isabell@stardust ~]$ cd /var/www/virtual/$USER/
  [isabell@stardust isabell]$ mkdir -p openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/
  [isabell@stardust isabell]$ touch openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/policy
  [isabell@stardust isabell]$ touch openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/.htaccess
@@ -77,14 +77,14 @@ Create the directories and symlinks
 
  [isabell@stardust ~]$ cd ~/
  [isabell@stardust ~]$ mkdir -p html/.well-known
- [isabell@stardust ~]$ ln -s /var/www/virtual/isabell/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/ /var/www/virtual/isabell/html/.well-known/openpgpkey
+ [isabell@stardust ~]$ ln -s /var/www/virtual/$USER/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/ /var/www/virtual/$USER/html/.well-known/openpgpkey
  [isabell@stardust ~]$
  
 Create another symlink to facilitate uploading keys via scp:
 
 ::
 
- [isabell@stardust ~]$ ln -s /var/www/virtual/isabell/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/ ./
+ [isabell@stardust ~]$ ln -s /var/www/virtual/$USER/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/ ./
  [isabell@stardust ~]$
 
 Upload GPG keys

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -194,6 +194,6 @@ Credits
 
 ----
 
-Tested with Uberspace 7.6.1.2 and gpg (GnuPG) 2.2.4
+Tested with gpg (GnuPG) 2.2.4, Uberspace 7.6.1.2
 
 .. author_list::

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -20,13 +20,13 @@ Web Key Directories provide an easy way to discover public keys through HTTPS. T
 
 .. note:: For this guide you should be familiar with the basic concepts of
 
-  * e-mail encryption with GnuPG / OpenPGP 
+  * e-mail encryption with GnuPG_ / OpenPGP_ 
   * :manual:`domains <web-domains>`
 
 Prerequisites
 =============
 
-You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@example.org). This guide does not explain how to generate GnuPG / OpenPGP keys.
+You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@example.org). This guide does not explain how to generate GnuPG_ / OpenPGP_ keys.
 
 The domain for the advanced method has to be set up at your DNS-Hoster and within your uberspace:
 
@@ -189,6 +189,8 @@ Credits
  * https://www.kuketz-blog.de/gnupg-web-key-directory-wkd-einrichten/ (German)
 
 .. _Web Key Directory: https://wiki.gnupg.org/WKD
+.. _GnuPG: https://gnupg.org/
+.. _OpenPGP: https://www.openpgp.org/software/
 
 ----
 

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -1,0 +1,174 @@
+.. highlight:: console
+
+.. author:: Starkstromkonsument <it@starkstromkonsument.de>
+
+.. tag:: audience-admins
+.. tag:: mail
+.. tag:: self-hosting
+
+#######
+Web Key Directory (WKD) for GnuPG
+#######
+
+.. tag_list::
+
+[WKD] stands for "`Web Key Directory`_" and is a standard for making a users public key available via their e-mail provider or server with the domain that corresponds to their e-mail address. There's several clients (such as Enigmail in Thunderbird) that will use this standard to automatically fetch a user's public key, when writing an e-mail to them.
+
+Web Key Directories provide an easy way to discover public keys through HTTPS. They provide an important piece to the infrastructure to improve the user experience for exchanging secure emails and files.
+
+In contrast to the public keyservers a Web Key Directory does not publish mail addresses. And it is an authoritative pubkey source for its domain.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * e-mail encryption with GnuPG / OpenPGP 
+  * :manual:`domains <web-domains>`
+
+Prerequisites
+=============
+
+You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@stardust.uberspace.de).
+
+The domain for the advanced method has to be set up at your DNS-Hoster and within your uberspace:
+
+::
+
+ [isabell@stardust ~]$ uberspace web domain list
+ openpgpkey.stardust.uberspace.de
+ [isabell@stardust ~]$
+
+
+Create Directory Structure
+==========================
+
+advanced method
+---------------
+
+The "advanced" method seems to be queried in front of the "direct" method by clients. If the subdomain openpgpkey resolves to a valid IP but the host at this IP does not provide the WKD (e.g. because of a wildcard in the DNS-Records), the WKD-lookup may fail.
+
+Create the directories and the empty files:
+
+::
+
+ [isabell@stardust ~]$ cd /var/www/virtual/isabell/
+ [isabell@stardust isabell]$ mkdir -p openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/
+ [isabell@stardust isabell]$ touch openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/policy
+ [isabell@stardust isabell]$ touch openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/.htaccess
+ [isabell@stardust ~]$
+
+Add the following lines to the .htaccess-file to set the correct headers:
+
+.. code-block:: apache2
+
+ ## WEB KEY DIRECTORY ##
+ <IfModule mod_mime.c>
+    ForceType application/octet-stream
+    Header always set Access-Control-Allow-Origin "*"
+ </IfModule>
+
+.. warning:: Make sure that there is no automatic directory listing! It is not necessary and it reveals the number of emailadresses (and their hashes) in the WKD (this can be a privacy issue)!
+
+
+direct method
+-------------
+
+Create the directories and symlinks
+
+::
+
+ [isabell@stardust ~]$ cd ~/
+ [isabell@stardust ~]$ mkdir -p html/.well-known
+ [isabell@stardust ~]$ ln -s /var/www/virtual/isabell/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/ /var/www/virtual/isabell/html/.well-known/openpgpkey
+ [isabell@stardust ~]$
+ 
+Create another symlink to facilitate uploading keys via scp:
+
+::
+
+ [isabell@stardust ~]$ ln -s /var/www/virtual/isabell/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/ ./
+ [isabell@stardust ~]$
+
+Upload GPG keys
+===============
+
+Obtaining the WKD-Hashes
+------------------------
+
+The keys are stored in files named by the WKD-Hashes of the Mailuser. Take the prefix of your e-mail address (i.e. in isabell@stardust.uberspace.de, this would be isabell), hash it with SHA-1 and then encode the output with z-base-32.
+
+You can get the hashes for all identities of your GPG key with this command:
+
+::
+
+ someuser@somehost ~]$ gpg --with-wkd-hash --list-public-keys "isabell@stardust.uberspace.de"
+ [...]
+       0123456789ABCDEF0123456789ABCDEF01234567
+ uid        [uneingeschränkt] Isabell <isabell@stardust.uberspace.de>
+            mmuhurigesr7z8hzf6sh5cmfsnmiiyyr@stardust.uberspace.de
+ [...]
+ someuser@somehost ~]$
+
+Alternatively you can use https://cryptii.com/pipes/z-base-32 for that (add a hash-block with SHA-1 before the z.base-32-block).
+
+Summary:
+
+ * E-Mail: isabell@startdust.uberspace.de
+ * Prefix: isabell
+ * WKD-Hash: mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+
+
+Exporting a GPG key
+-------------------
+
+.. note:: The public key is the the same for all identities of a key. You can simply duplicate the exported key, using the corresponding WKD-Hash as filename.
+
+::
+
+ [someuser@somehost ~]$ gpg --no-armor --export isabell@stardust.uberspace.de > mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+ [someuser@somehost ~]$
+
+Upload
+------
+
+::
+
+ [someuser@somehost ~]$ scp mmuhurigesr7z8hzf6sh5cmfsnmiiyyr isabell@stardust.uberspace.de:hu/
+ [someuser@somehost ~]$
+
+Testing
+=======
+
+You can test the download by running:
+
+::
+
+ [someuser@somehost ~]$ env GNUPGHOME=$(mktemp -d) gpg --locate-keys --auto-key-locate clear,wkd,nodefault isabell@stardust.uberspace.de
+ gpg: keybox '/tmp/tmp.c8iW067tlp/pubring.kbx' created
+ gpg: /tmp/tmp.c8iW067tlp/trustdb.gpg: trustdb created
+ gpg: key 89ABCDEF01234567: public key "Isabell <isabell@stardust.uberspace.de>" imported
+ gpg: Total number processed: 1
+ gpg:               imported: 1
+ [...]
+       0123456789ABCDEF0123456789ABCDEF01234567
+ uid           [ unknown] Isabell <isabell@stardust.uberspace.de>
+ [...]
+ [someuser@somehost ~]$
+
+Alternatively you can use this Web-Tool: https://metacode.biz/openpgp/web-key-directory
+
+Credits
+=======
+
+ * https://wiki.gnupg.org/WKDHosting
+ * https://spacekookie.de/blog/usable-gpg-with-wkd/
+ * https://www.kuketz-blog.de/gnupg-web-key-directory-wkd-einrichten/ (German)
+
+
+.. _Web Key Directory: https://wiki.gnupg.org/WKD
+
+----
+
+Tested with Uberspace 7.6.0.0
+
+.. author_list::

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -192,6 +192,6 @@ Credits
 
 ----
 
-Tested with Uberspace 7.6.0.0
+Tested with Uberspace 7.6.1.2 and gpg (GnuPG) 2.2.4
 
 .. author_list::

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -26,14 +26,14 @@ Web Key Directories provide an easy way to discover public keys through HTTPS. T
 Prerequisites
 =============
 
-You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@stardust.uberspace.de). This guide does not explain how to generate GnuPG / OpenPGP keys.
+You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@example.org). This guide does not explain how to generate GnuPG / OpenPGP keys.
 
 The domain for the advanced method has to be set up at your DNS-Hoster and within your uberspace:
 
 ::
 
  [isabell@stardust ~]$ uberspace web domain list
- openpgpkey.stardust.uberspace.de
+ openpgpkey.example.org
  [isabell@stardust ~]$
 
 
@@ -50,9 +50,9 @@ Create the directories and the empty files:
 ::
 
  [isabell@stardust ~]$ cd /var/www/virtual/$USER/
- [isabell@stardust isabell]$ mkdir -p openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/
- [isabell@stardust isabell]$ touch openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/policy
- [isabell@stardust isabell]$ touch openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/.htaccess
+ [isabell@stardust isabell]$ mkdir -p openpgpkey.example.org/.well-known/openpgpkey/example.org/hu/
+ [isabell@stardust isabell]$ touch openpgpkey.example.org/.well-known/openpgpkey/example.org/policy
+ [isabell@stardust isabell]$ touch openpgpkey.example.org/.well-known/openpgpkey/example.org/.htaccess
  [isabell@stardust ~]$
 
 Add the following lines to the empty ``.htaccess`` file to set the correct headers:
@@ -77,14 +77,14 @@ Create the directories and symlinks
 
  [isabell@stardust ~]$ cd ~/
  [isabell@stardust ~]$ mkdir -p html/.well-known
- [isabell@stardust ~]$ ln -s /var/www/virtual/$USER/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/ /var/www/virtual/$USER/html/.well-known/openpgpkey
+ [isabell@stardust ~]$ ln -s /var/www/virtual/$USER/openpgpkey.example.org/.well-known/openpgpkey/example.org/ /var/www/virtual/$USER/html/.well-known/openpgpkey
  [isabell@stardust ~]$
  
 Create another symlink to facilitate uploading keys via scp:
 
 ::
 
- [isabell@stardust ~]$ ln -s /var/www/virtual/$USER/openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/ ./
+ [isabell@stardust ~]$ ln -s /var/www/virtual/$USER/openpgpkey.example.org/.well-known/openpgpkey/example.org/hu/ ./
  [isabell@stardust ~]$
 
 Upload GPG keys
@@ -93,24 +93,24 @@ Upload GPG keys
 Obtaining the WKD-Hashes
 ------------------------
 
-The keys are stored in files named by the WKD-Hashes of the Mailuser. Take the prefix of your e-mail address (i.e. in isabell@stardust.uberspace.de, this would be `isabell`), hash it with SHA-1 and then encode the output with z-base-32.
+The keys are stored in files named by the WKD-Hashes of the Mailuser. Take the prefix of your e-mail address (i.e. in isabell@example.org, this would be `isabell`), hash it with SHA-1 and then encode the output with z-base-32.
 
 You can get the hashes for all identities of your GPG key with this command:
 
 ::
 
- [someuser@somehost ~]$ gpg --with-wkd-hash --list-public-keys "isabell@stardust.uberspace.de"
+ [someuser@somehost ~]$ gpg --with-wkd-hash --list-public-keys "isabell@example.org"
  [...]
        0123456789ABCDEF0123456789ABCDEF01234567
- uid           [ unknown] Isabell <isabell@stardust.uberspace.de>
-            mmuhurigesr7z8hzf6sh5cmfsnmiiyyr@stardust.uberspace.de
+ uid           [ unknown] Isabell <isabell@example.org>
+            mmuhurigesr7z8hzf6sh5cmfsnmiiyyr@example.org
  [...]
  [someuser@somehost ~]$
 
 Alternatively you can use https://cryptii.com/pipes/z-base-32 for that (add a hash-block with SHA-1 before the z.base-32-block).
 
 Summary:
- * E-Mail: isabell@startdust.uberspace.de
+ * E-Mail: isabell@example.org
  * Prefix: isabell
  * WKD-Hash: mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
 
@@ -120,7 +120,7 @@ Exporting a GPG key
 
 ::
 
- [someuser@somehost ~]$ gpg --no-armor --export isabell@stardust.uberspace.de > mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+ [someuser@somehost ~]$ gpg --no-armor --export isabell@example.org > mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
  [someuser@somehost ~]$
  
 .. note:: The public key is the the same for all identities of a key. You can simply duplicate the exported key, using the corresponding WKD-Hash as filename.
@@ -130,7 +130,7 @@ Upload
 
 ::
 
- [someuser@somehost ~]$ scp mmuhurigesr7z8hzf6sh5cmfsnmiiyyr isabell@stardust.uberspace.de:hu/
+ [someuser@somehost ~]$ scp mmuhurigesr7z8hzf6sh5cmfsnmiiyyr isabell@example.org:hu/
  [someuser@somehost ~]$
 
 Testing
@@ -138,15 +138,15 @@ Testing
 
 First of all, these two URLs should be available using your browser:
 
- * Advanced method: https://openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
- * Direct method: https://stardust.uberspace.de/.well-known/openpgpkey/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+ * Advanced method: https://openpgpkey.example.org/.well-known/openpgpkey/example.org/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+ * Direct method: https://example.org/.well-known/openpgpkey/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
 
 or ``curl``:
 
 .. code-block:: console
  :emphasize-lines: 4,6
 
- [someuser@somehost ~]$ curl -I https://openpgpkey.stardust.uberspace.de/.well-known/openpgpkey/stardust.uberspace.de/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
+ [someuser@somehost ~]$ curl -I https://openpgpkey.example.org/.well-known/openpgpkey/example.org/hu/mmuhurigesr7z8hzf6sh5cmfsnmiiyyr
  HTTP/2 200 
  date: Sat, 02 May 2020 19:16:17 GMT
  content-type: application/octet-stream
@@ -167,15 +167,15 @@ You can test the WKD-download by running:
 
 ::
 
- [someuser@somehost ~]$ env GNUPGHOME=$(mktemp -d) gpg --locate-keys --auto-key-locate clear,wkd,nodefault isabell@stardust.uberspace.de
+ [someuser@somehost ~]$ env GNUPGHOME=$(mktemp -d) gpg --locate-keys --auto-key-locate clear,wkd,nodefault isabell@example.org
  gpg: keybox '/tmp/tmp.c8iW067tlp/pubring.kbx' created
  gpg: /tmp/tmp.c8iW067tlp/trustdb.gpg: trustdb created
- gpg: key 89ABCDEF01234567: public key "Isabell <isabell@stardust.uberspace.de>" imported
+ gpg: key 89ABCDEF01234567: public key "Isabell <isabell@example.org>" imported
  gpg: Total number processed: 1
  gpg:               imported: 1
  [...]
        0123456789ABCDEF0123456789ABCDEF01234567
- uid           [ unknown] Isabell <isabell@stardust.uberspace.de>
+ uid           [ unknown] Isabell <isabell@example.org>
  [...]
  [someuser@somehost ~]$
 

--- a/source/guide_wkd.rst
+++ b/source/guide_wkd.rst
@@ -26,7 +26,7 @@ Web Key Directories provide an easy way to discover public keys through HTTPS. T
 Prerequisites
 =============
 
-You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@example.org). This guide does not explain how to generate GnuPG_ / OpenPGP_ keys.
+You need to own a domain (e.g. for example.org) and to have the ready to use public keys for the e-mail identities you want to provide via WKD (e.g. isabell@example.org). This guide does not explain how to generate GnuPG_ / OpenPGP_ keys.
 
 The domain for the advanced method has to be set up at your DNS-Hoster and within your uberspace:
 

